File: plugins/plugin/buttons/UnarmedAttackTypeButtons.java
package plugin.buttons;

import astraeus.game.event.SubscribesTo;
import astraeus.game.event.impl.ButtonActionEvent;
import astraeus.game.model.entity.mob.player.Player;
import astraeus.net.packet.out.SetWidgetConfigPacket;

@SubscribesTo(ButtonActionEvent.class)
public class UnarmedAttackTypeButtons extends ButtonClick {

	@Override
	protected void execute(Player player, ButtonActionEvent event) {
		switch (event.getButton()) {
		
		case 12464:
	        player.attr().toggle(Player.ACCEPT_AID_KEY);
	        player.queuePacket(new SetWidgetConfigPacket(427, player.attr().get(Player.ACCEPT_AID_KEY)));
			break;
		
		}
	}

	@Override
	public boolean test(ButtonActionEvent event) {
		return event.getButton() == 12464;
	}

}

File: plugins/plugin/buttons/UnarmedAttackTypeButtons.kt
package plugin.buttons

import astraeus.game.event.SubscribesTo
import astraeus.game.event.impl.ButtonActionEvent
import astraeus.game.model.entity.mob.combat.def.AttackType
import astraeus.game.model.entity.mob.player.Player

@SubscribesTo(ButtonActionEvent::class)
class UnarmedAttackTypeButtons : ButtonClick() {

	override fun execute(player: Player, event: ButtonActionEvent) {
		when (event.getButton()) {
			5860 -> player.setAttackType(AttackType.UNARMED_PUNCH)
			5861 -> player.setAttackType(AttackType.UNARMED_BLOCK)
			5862 -> player.setAttackType(AttackType.UNARMED_KICK)
		}
	}


	override fun test(event: ButtonActionEvent): Boolean {
		if (event.getButton() === 5860 || event.getButton() === 5861 || event.getButton() === 5862) {
			return true
		}
		return false
	}
	
	
}